src/service/mercury/index.ts

```diff
@@ -644,6 +649,46 @@ export class MercuryClient {
       network,
       isFailedIncluded,
     );
+
+    if (horizonResponse.data && Array.isArray(horizonResponse.data)) {
```

Comment:
we could add `&& shouldScan` directly here to remove 1 indentation from the code block